tests/versioned/apollo-federation/query-obfuscation.test.js

```diff
@@ -43,26 +37,28 @@ function createQueryObfuscaionTests(t) {
 
     const path = 'library.booksInStock.title'
 
+    let tx
     helper.agent.on('transactionFinished', (transaction) => {
+      tx = transaction
+    })
```

Comment:
I think this pattern is going to be mysterious to anyone reading these tests in the future. But it would also be too verbose to put a comment on every instance, and too hidden to put it on only a single instance. So... I have no idea how to solve that problem.

Reply:
Agreed. I went down that path and backed out. It's just that due to how our agent instruments and how these tests make http requests and assert while http request is still ongoing could be masking the issue. It also has me thinking why is this the only library exhibiting this kind of behavior.